src/openvic-simulation/map/Map.cpp

```diff
@@ -670,3 +670,82 @@ bool Map::generate_and_load_province_adjacencies(std::vector<LineObject> const&
 	);
 	return ret;
 }
+
+bool Map::load_climate_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid climate identifier - empty!");
+			return false;
+		}
+
+		Climate* cur_climate = climates.get_item_by_identifier(identifier);
+		if (cur_climate == nullptr) {
+			ModifierValue values;
+			bool ret = modifier_manager.expect_modifier_value(move_variable_callback(values))(node);
+			ret &= climates.add_item({ identifier, std::move(values) });
+			return ret;
+		} else {
+			bool ret = expect_list([this, &identifier, &cur_climate](ast::NodeCPtr list_element) -> bool {
+				std::string_view cur_province_id;
+				bool ret = expect_identifier_or_string(assign_variable_callback(cur_province_id))(list_element);
+				Province* cur_province = provinces.get_item_by_identifier(cur_province_id);
+				if (cur_province != nullptr) {
+					cur_climate->add_province(cur_province);
+					return ret;
+				} else {
+					Logger::error("Invalid province id ", cur_province_id, " in definition of climate ", identifier, "!");
+					return false;
+				}
+			})(node);
+			return ret;
+		}
+	})(root);
+
+	lock_climates();
+
+	for (Climate const& cl : climates.get_items()) {
+		for(Province* prov : cl.get_provinces()) {
+			prov->climate = &cl;
+		}
+	}
+
+	return ret;
+}
+
+bool Map::load_continent_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
```

Comment:
```suggestion
	bool ret = expect_dictionary_reserve_length(continents, [this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
```
Since each entry is a new continent, we can reserve space in `continents` using the number of entries in the file.